src/http/http_endpoint.h

```diff
@@ -174,6 +183,46 @@ namespace http
       send_raw(std::move(data));
     }
 
+    void record_response_txid(std::span<const uint8_t> raw_response) override
+    {
+      // To avoid a full HTTP parse, search for the desired header directly
+      std::string_view response(
+        (char const*)raw_response.data(), raw_response.size());
+      std::string_view target(http::headers::CCF_TX_ID);
+      auto header_begin = std::search(
+        response.begin(), response.end(), target.begin(), target.end());
+      auto header_name_end = std::find(header_begin, response.end(), ':');
+      auto header_value_end = std::find(header_name_end, response.end(), '\r');
```

Comment:
That seems unfortunate at best, and possibly insecure at worst if it's possible to add CCF_TX_ID inline as a header value, followed by a bogus value. We should at least look for `"\r{}", CCF_TX_ID`, I think.